src/ssh/azext_ssh/target_properties_utils.py

```diff
@@ -0,0 +1,142 @@
+    from .aaz.latest.connected_v_mwarev_sphere.virtual_machine import Show as VMwarevSphereShow
+
+    # pylint: disable=broad-except
+    try:
+        get_args = {'resource_group': resource_group_name,'machine_name': vm_name}
+        # Retrieves the Connected VMware VM object
+        return VMwarevSphereShow(cli_ctx=cmd.cli_ctx)(command_args=get_args)
+    except Exception:
+        return None
+
+
+def parse_os_type(properties, resource_type):
+    if resource_type == "microsoft.compute/virtualmachines":
+        if properties.storage_profile and properties.storage_profile.os_disk and properties.storage_profile.os_disk.os_type:
+            return properties.storage_profile.os_disk.os_type
+        
+    elif resource_type == "microsoft.hybridcompute/machines":
+        if properties.get('osName', None):
+            return properties.get('osName')
+        elif properties.get('properties', None):
+            return properties.get('properties').get('osType', None)
+        
+    elif resource_type == "microsoft.connectedvmwarevsphere/virtualmachines":
+        if properties.get("osProfile") and properties.get("osProfile").get("osType"):
+            return properties.get("osProfile").get("osType")
+
+
+def parse_agent_version(properties, resource_type):
+    if resource_type == "microsoft.compute/virtualmachines":
+        return None
+        
+    elif resource_type == "microsoft.hybridcompute/machines":
+        if properties.get('properties'):
+            return properties.get('properties').get('agentVersion')
+        
+    elif resource_type == "microsoft.connectedvmwarevsphere/virtualmachines":
+        if properties.get("properties") and properties.get("properties").get('guestAgentProfile') and\
+            properties.get("properties").get('guestAgentProfile').get('agentVersion'):
+            return properties.get("properties").get('guestAgentProfile').get('agentVersion')
+    
+
+def parse_tags(properties, resource_type):
+    if resource_type == "microsoft.compute/virtualmachines":
+        if properties.tags:
+            return properties.tags
+        
+    else:
+        if properties.get('tags'):
+            return properties.get('tags')
+        
+
+# This function is used to check if the OS type is valid and if the authentication options are valid for that OS
+def check_valid_os_type(os_type, agent_version, op_info):
+    if os_type:
+        logger.debug("Target OS Type: %s", os_type)
+        telemetry.add_extension_event('ssh', {'Context.Default.AzureCLI.TargetOSType': os_type})
+
+    if os_type and os_type.lower() == 'windows' and not op_info.local_user:
+        colorama.init()
+        error_message = "SSH Login using AAD credentials is not currently supported for Windows."
+        recommendation = colorama.Fore.YELLOW + "Please provide --local-user." + colorama.Style.RESET_ALL
+        raise azclierror.RequiredArgumentMissingError(error_message, recommendation)
+
+    if op_info.is_arc() and agent_version:
```

Comment:
You should separate the handling the agent version into a different function, separate from check_valid_os_type, because the two functionalities are not related. Just create a check_valid_agent_version or something like that just for this check.